test/integration/telemetry_test.exs

```diff
@@ -131,6 +122,19 @@ defmodule Oban.Integration.TelemetryTest do
     :telemetry.detach("oban-default-logger")
   end
 
+  test "span publishes given metric and metadata" do
+    :telemetry.attach("span-handler", [:oban, :test_metric], &Handler.handle/4, self())
+
+    assert :example_return = Telemetry.span(:test_metric, fn -> :example_return end)
+    assert :ok = Telemetry.span(:test_metric, fn -> :timer.sleep(5) end, %{example_metadata: 1})
```

Comment:
Functionally they are identical, but `Process.sleep/1` is used everywhere else in the codebase:

```suggestion
    assert :ok = Telemetry.span(:test_metric, fn -> Process.sleep(5) end, %{example_metadata: 1})
```